crates/bevy_ui/src/render/mod.rs

```diff
@@ -522,125 +529,72 @@ pub fn extract_uinode_borders(
-            },
-        );
-    }
-}
-
-pub fn extract_uinode_outlines(
-    mut commands: Commands,
-    mut extracted_uinodes: ResMut<ExtractedUiNodes>,
-    default_ui_camera: Extract<DefaultUiCamera>,
-    uinode_query: Extract<
-        Query<(
-            &Node,
-            &GlobalTransform,
-            &ViewVisibility,
-            Option<&CalculatedClip>,
-            Option<&TargetCamera>,
-            &Outline,
-        )>,
-    >,
-) {
-    let image = AssetId::<Image>::default();
-    for (node, global_transform, view_visibility, maybe_clip, camera, outline) in &uinode_query {
-        let Some(camera_entity) = camera.map(TargetCamera::entity).or(default_ui_camera.get())
-        else {
-            continue;
-        };
-
-        // Skip invisible outlines
-        if !view_visibility.get()
-            || outline.color.is_fully_transparent()
-            || node.outline_width == 0.
+        // don't extract border if no border or the node is zero-sized (a zero sized node can have an outline).
+        if uinode.size().x > 0.
+            && uinode.size().y > 0.
+            && !(left == 0.0 && top == 0.0 && right == 0.0 && bottom == 0.0)
         {
-            continue;
-        }
-
-        // Calculate the outline rects.
-        let inner_rect = Rect::from_center_size(Vec2::ZERO, node.size() + 2. * node.outline_offset);
-        let outer_rect = inner_rect.inflate(node.outline_width());
-        let outline_edges = [
-            // Left edge
-            Rect::new(
-                outer_rect.min.x,
-                outer_rect.min.y,
-                inner_rect.min.x,
-                outer_rect.max.y,
-            ),
-            // Right edge
-            Rect::new(
-                inner_rect.max.x,
-                outer_rect.min.y,
-                outer_rect.max.x,
-                outer_rect.max.y,
-            ),
-            // Top edge
-            Rect::new(
-                inner_rect.min.x,
-                outer_rect.min.y,
-                inner_rect.max.x,
-                inner_rect.min.y,
-            ),
-            // Bottom edge
-            Rect::new(
-                inner_rect.min.x,
-                inner_rect.max.y,
-                inner_rect.max.x,
-                outer_rect.max.y,
-            ),
-        ];
-
-        let world_from_local = global_transform.compute_matrix();
-        for edge in outline_edges {
-            if edge.min.x < edge.max.x && edge.min.y < edge.max.y {
+            if let Some(border_color) = maybe_border_color {
+                let transform = global_transform.compute_matrix();
                 extracted_uinodes.uinodes.insert(
                     commands.spawn_empty().id(),
                     ExtractedUiNode {
-                        stack_index: node.stack_index,
+                        stack_index: uinode.stack_index,
                         // This translates the uinode's transform to the center of the current border rectangle
```

Comment:
This comment is slightly misplaced now. Actually, it would be nice if the variable had a descriptive name instead.